cypress/integration/cucumber/step_definitions/refapp-3.x/06-patient-search/patient-search.js

```diff
@@ -0,0 +1,41 @@
+import {Given} from 'cypress-cucumber-preprocessor/steps';
+
+let patient_uuid = null;
+
+beforeEach("Take a dump of the database and restore",() => {
```

Comment:
How about `Restore Database instead` of `Take a dump of database and restore`

Reply:
yeah this is good @pasindur99 doest it make sense to you

Reply:
We should really create the database dump in a `beforeAll()` hook and then in the `beforeEach()` just restore the database state (and fail if the file to restore from doesn't exist). Also, I think it's a little odd to use an NPM script to call a Bash script... It seems cleaner to just call the bash script directly via `cy.exec()` (there are fewer processes, it uses fewer resources, there's less indirection...)

Reply:
Yeah, @jnsereko let's rename the heading.

Reply:
Thank you for the suggestion @ibacher. 

-  I tested the bash command by using it directly in `cy.exec()`. So there are no issues with that. We can use the command directly. cc: @jnsereko 
- By using `beforeEach()` hook, we could make a dump of the database at the initial scenario of the test running. After the initial test was executed the other test will also use the `beforeEach()` to restore the database since we have both commands inside an if condition. If there is no dump file it means the tests are in the initial stage. Then it makes the dump there. If there is a dump file it means that will be a test case afterward. So the database will restore at that point. And the database is a snapshot of the initial stage. So is there any need of using two hooks to make a dump and restore the database?  @ibacher

Reply:
Agree with @pasindur99. We need to take the dump only after creating a new openmrs instance (to take a fresh copy of the initial DB).